activerecord/lib/active_record/database_configurations/database_config.rb

```diff
@@ -13,6 +13,11 @@ def initialize(env_name, spec_name)
         @spec_name = spec_name
       end
 
+      def config
```

Comment:
This object is private API. Do we need to deprecate and have a Changelog entry?

Reply:
This object is private but the other two HashConfig and UrlConfig which inherit from this aren't. They shouldn't be used directly but when you loop through the configurations it will call config and 💥 

Some apps may be looping through the configs and then calling config on it. I know we are in GitHub - and eventually all that code will go away as we remove the multi-db stuff but it's not impossible someone is relying on it.

```
ActiveRecord::Base.configurations.configs_for(env_name: "development").each do |db_config|
db_config.config # this would blow up
end
```

Reply:
Oops I totally wrote the wrong thing here. I'm going back and forth between our app and Rails work and I dunno. 

The reason I want to deprecate the string access is because I know at least our app depends on it. It's likely apps are doing `db_config.config["database"]` because they used to access that with a hash. I'm more comfortable deprecating than potentially breaking apps and gems that rely on the old hash behavior. Especially as we move away from hashes internally (and maybe one day in apps themselves).

Reply:
👍